Libraries/Image/RCTLocalAssetImageLoader.m

```diff
@@ -50,7 +51,49 @@ - (RCTImageLoaderCancellationBlock)loadImageForURL:(NSURL *)imageURL
     }
 
     NSString *imageName = RCTBundlePathForURL(imageURL);
-    UIImage *image = [UIImage imageNamed:imageName];
+    NSBundle *targetBundle = [NSBundle mainBundle];
+    
+    if (bundlePath) {
+      NSString *targetBundleName = RCTBundlePathForURL([NSURL URLWithString:bundlePath]);
+      NSString *targetBundleNameWithExtension = [NSString stringWithFormat:@"%@%@", targetBundleName, @".bundle"];
+      NSMutableArray *bundles = [[NSBundle allBundles] mutableCopy];
+      
+      // Get any bundle files included in the mainBundle and add them to the array
+      for (NSString *path in [[NSBundle mainBundle] pathsForResourcesOfType:@"bundle" inDirectory:nil]) {
```

Comment:
Added a map to keep track of bundles which have been loaded; while they can't change at runtime they can change between loads.

It also doesn't really need to be more complicated than this because (as far as I can tell) it doesn't load the entire bundle in to memory when storing it.